plone/namedfile/tests/test_scaling.py

```diff
@@ -980,6 +980,34 @@ def test_title(self):
         )
 
 
+class Img2PictureTagTests(unittest.TestCase):
+    """Low level tests for Img2PictureTag."""
+
+    def _makeOne(self):
+        return plone.namedfile.picture.Img2PictureTag()
+
+    def test_update_src_scale(self):
+        update_src_scale = self._makeOne().update_src_scale
+        self.assertEqual(
+            update_src_scale("foo/fieldname/old", "new"),
+            "foo/fieldname/new"
+        )
+        self.assertEqual(
+            update_src_scale("@@images/fieldname/old", "mini"),
+            "@@images/fieldname/mini"
+        )
+        self.assertEqual(
+            update_src_scale("@@images/fieldname", "preview"),
```

Comment:
@mauritsvanrees  If the src scale wasn't there and it used the full size picture (the original). shouldn't it always be changed to use scale 'original' or alternatively the scale 'huge' (since we don't upscale images to a scale, huge will also be the orginal size, but with an upper cap).   

The problem there is assuming there is a scale 'orginal' or 'huge' registered.....

Reply:
I would say it is still fine for a rich text field to point to the original image, even though I don't see a way to do this with the standard TinyMCE setup.  Could be done as improvement.

No `original` scale exists.  This might be something to add as a default built-in available scale, but this could also confuse things, having a second way to do the same thing.

We could look through the available scales and/or picture variants and use the largest one.

But in all, I would say the way I do it in this PR is fine.